src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java

```diff
@@ -1,24 +1,62 @@
 package com.spanish_inquisition.battleship.client;
 
+import com.spanish_inquisition.battleship.client.network.SocketClient;
 import javafx.fxml.FXML;
 import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.VBox;
+
+import java.io.IOException;
+import java.util.logging.Level;
+
+import static com.spanish_inquisition.battleship.common.AppLogger.logger;
 
 /**
  * @author Michal_Partacz
  * A controller for a fxml file placed in resources folder. The methods annotated with @FXML react to events specified
  * in the file's UI elements.
  */
 public class MainMenuController {
-
     public Button featureButton;
+    public Label gameStatusLabel;
+    public TextField nameTextField;
+    public VBox centralVBox;
+
+    public SocketClient socketClient;
+
     /**
      * This method is run automatically right after the fxml file's loaded
      */
     @FXML
-    public void initialize() {}
+    public void initialize() {
+        try {
+            this.socketClient = SocketClient.createSocketClientWithSocket();
```

Comment:
It is, though it is not needed in the Controller class, let's leave the socket stuff in dedicated socket class.